presentation-compiler/src/main/dotty/tools/pc/completions/Completions.scala

```diff
@@ -535,7 +532,7 @@ class Completions(
     val query = completionPos.query
     if completionMode.is(Mode.Scope) && query.nonEmpty then
       val visitor = new CompilerSearchVisitor(sym =>
-        if Completion.isValidCompletionSymbol(sym, completionMode) &&
+        if Completion.isValidCompletionSymbol(sym, completionMode, isNew) &&
```

Comment:
I feel like maybe it would be a bit more natural to pass untyped tree here instead of `isNew`, since `isNew` is always calculated via the same function and can be postponed.

Reply:
We would compute it on every `isValidSymbol` call, and it can be hundreds of calls, tho ? I did especially this way because we don't recompute this check

Reply:
Oh, I see, makes sense.

Reply:
There is still duplication, as we're first filtering them in the compiler completions, and then again in Completions.scala from presentation compiler, but I think that this check is fast enough, that eliminating completions before searching for apply / constructors in `withCompletionSuffix` is completely worth it.